frontend/src/lib/components/LemonCalendarRange/LemonCalendarRangeInline.tsx

```diff
@@ -0,0 +1,126 @@
+import { LemonCalendar } from 'lib/components/LemonCalendar/LemonCalendar'
+import React, { useEffect, useState } from 'react'
+import { dayjs } from 'lib/dayjs'
+import clsx from 'clsx'
+import { LemonCalendarRangeProps } from './LemonCalendarRange'
+
+/** Used to calculate how many calendars fit on the screen */
+const WIDTH_OF_ONE_CALENDAR_MONTH = 300
+/** Number of calendars to display if `typeof window === undefined` */
+const CALENDARS_IF_NO_WINDOW = 2
+
+export function LemonCalendarRangeInline({
+    value,
+    onChange,
+    months,
+}: Omit<LemonCalendarRangeProps, 'onClose'>): JSX.Element {
+    // Keep a sanitised and cached copy of the selected range
+    const [valueStart, valueEnd] = [
+        value?.[0] ? dayjs(value[0]).format('YYYY-MM-DD') : null,
+        value?.[1] ? dayjs(value[1]).format('YYYY-MM-DD') : null,
+    ]
+    const [[rangeStart, rangeEnd, lastChanged], _setRange] = useState([valueStart, valueEnd, 'end' as 'start' | 'end'])
+
+    function setRange([rangeStart, rangeEnd, lastChanged]: [string | null, string | null, 'start' | 'end']): void {
+        _setRange([rangeStart, rangeEnd, lastChanged])
+        if (rangeStart && rangeEnd) {
+            onChange([rangeStart, rangeEnd])
+        }
+    }
+
+    // How many months fit on the screen, capped between 1..2
+    function getMonthCount(): number {
+        const width =
+            typeof window === undefined ? WIDTH_OF_ONE_CALENDAR_MONTH * CALENDARS_IF_NO_WINDOW : window.innerWidth
+        return Math.min(Math.max(1, Math.floor(width / WIDTH_OF_ONE_CALENDAR_MONTH)), 2)
+    }
+    const [autoMonthCount, setAutoMonthCount] = useState(getMonthCount())
+    useEffect(() => {
+        const listener = (): void => {
+            if (autoMonthCount !== getMonthCount()) {
+                setAutoMonthCount(getMonthCount())
+            }
+        }
+        window.addEventListener('resize', listener)
+        return () => window.removeEventListener('resize', listener)
+    }, [autoMonthCount])
```

Comment:
Hmm, this auto month code doesn't work at all, not now on app, nor in this PR. I'll happily fix this after the PR.

However, this could go into a hook `const autoMonthCount = useAutoMonthCount()`, which can be shared between the inline and popup components to retain the previous behaviour regarding showing the selected range.